code/modules/awaymissions/corpse.dm

```diff
@@ -4,294 +4,528 @@
-	corpsesuit = /obj/item/clothing/suit/storage/labcoat/science
-	corpseback = /obj/item/storage/backpack
-	corpseshoes = /obj/item/clothing/shoes/white
-	corpseid = 1
-	corpseidjob = "Scientist"
-	corpseidaccess = "Scientist"
-
-/obj/effect/landmark/corpse/miner
-	corpseradio = /obj/item/radio/headset/headset_cargo
-	corpseuniform = /obj/item/clothing/under/rank/miner
-	corpsegloves = /obj/item/clothing/gloves/fingerless
-	corpseback = /obj/item/storage/backpack/industrial
-	corpseshoes = /obj/item/clothing/shoes/black
-	corpseid = 1
-	corpseidjob = "Shaft Miner"
-	corpseidaccess = "Shaft Miner"
-
-/obj/effect/landmark/corpse/miner/hardsuit
-	corpsesuit = /obj/item/clothing/suit/space/hardsuit/mining
-	corpsemask = /obj/item/clothing/mask/breath
-	corpsehelmet = /obj/item/clothing/head/helmet/space/hardsuit/mining
+	id_job = "Scientist"
+	outfit = /datum/outfit/job/scientist
+
+/obj/effect/mob_spawn/human/miner
+	name = "Shaft Miner"
+	id_job = "Shaft Miner"
+	outfit = /datum/outfit/job/mining/suit
+
+/datum/outfit/job/mining/suit
+	name = "Station Engineer"
+	suit = /obj/item/clothing/suit/space/hardsuit/mining
+	head = /obj/item/clothing/head/helmet/space/hardsuit/mining
+	uniform = /obj/item/clothing/under/rank/miner
+	gloves = /obj/item/clothing/gloves/fingerless
+	shoes = /obj/item/clothing/shoes/workboots
+	l_ear = /obj/item/radio/headset/headset_cargo/mining
+	id = /obj/item/card/id/supply
+	l_pocket = /obj/item/reagent_containers/food/pill/patch/styptic
+	r_pocket = /obj/item/flashlight/seclite
+
+
+/obj/effect/mob_spawn/human/bartender
+	name = "Space Bartender"
+	id_job = "Bartender"
+	id_access_list = list(access_bar)
+	outfit = /datum/outfit/spacebartender
+
+/obj/effect/mob_spawn/human/bartender/alive
+	death = FALSE
+	roundstart = FALSE
+	random = TRUE
+	name = "bartender sleeper"
+	icon = 'icons/obj/Cryogenic2.dmi'
+	icon_state = "sleeper"
+	flavour_text = "You are a space bartender!"
+	assignedrole = "Space Bartender"
+
+/datum/outfit/spacebartender
+	name = "Space Bartender"
+	uniform = /obj/item/clothing/under/rank/bartender
+	suit = /obj/item/clothing/suit/armor/vest
+	belt = /obj/item/storage/belt/bandolier/full
+	shoes = /obj/item/clothing/shoes/black
+	glasses = /obj/item/clothing/glasses/sunglasses/reagent
+	id = /obj/item/card/id
 
 
+/obj/effect/mob_spawn/human/beach
+	outfit = /datum/outfit/beachbum
+
+/obj/effect/mob_spawn/human/beach/alive
+	death = FALSE
+	roundstart = FALSE
+	random = TRUE
+	mob_name = "Beach Bum"
+	name = "beach bum sleeper"
+	icon = 'icons/obj/Cryogenic2.dmi'
+	icon_state = "sleeper"
+	flavour_text = "You are a beach bum!"
+	assignedrole = "Beach Bum"
+
+/datum/outfit/beachbum
+	name = "Beach Bum"
+	glasses = /obj/item/clothing/glasses/sunglasses
+	uniform = /obj/item/clothing/under/shorts/red
+
+/////////////////Spooky Undead//////////////////////
+
+/obj/effect/mob_spawn/human/skeleton
+	name = "skeletal remains"
+	mob_name = "skeleton"
+	mob_species = "Skeleton"

```

Comment:
make this the datum path. Also make sure that all `mob_species` is a datum path and not text path.